### PR TITLE
net-misc/icingadb: add 1.2.0 with icingadb-migrate and systemd unit

### PR DESCRIPTION
**Pull request by sbraz**
* Add a new USE flag that controls the build and installation of the icingadb-migrate tool.
* Add a systemd unit.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by sbraz** [comment]
Diff to the previous ebuild. If someone has a better name for the USE flag, I'm all ears :)
```diff
--- icingadb-1.1.1.ebuild       2024-07-06 22:45:06.419461825 +0200
+++ icingadb-1.2.0.ebuild       2024-07-06 22:47:02.160297055 +0200
@@ -3,16 +3,19 @@
 
 EAPI=8
 
-inherit go-module
+inherit go-module systemd
 
 DESCRIPTION="Icinga configuration and state database supporting multiple environments"
 HOMEPAGE="https://icinga.com/docs/icinga-db/"
-SRC_URI="https://github.com/Icinga/icingadb/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
-       https://dev.gentoo.org/~juippis/distfiles/${P}-deps.tar.xz"
+SRC_URI="
+       https://github.com/Icinga/icingadb/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+       https://dev.gentoo.org/~sbraz/distfiles/${P}-deps.tar.xz
+"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="amd64 ~arm64"
+KEYWORDS="~amd64 ~arm64"
+IUSE="migrate"
 
 RDEPEND="
        acct-user/icinga
@@ -21,16 +24,30 @@
 
 DOCS=( config.example.yml doc README.md schema )
 
+src_prepare() {
+       default
+       sed -e "s|@EPREFIX@|${EPREFIX}|" "${FILESDIR}/icingadb.service" > "${WORKDIR}/icingadb.service" || die
+}
+
 src_compile() {
-       cd cmd/icingadb || die
+       pushd cmd/icingadb || die
        ego build
+       if use migrate; then
+               popd || die
+               pushd cmd/icingadb-migrate || die
+               ego build
+       fi
 }
 
 src_install() {
        dobin cmd/icingadb/icingadb
+       if use migrate; then
+               dobin cmd/icingadb-migrate/icingadb-migrate
+       fi
        einstalldocs
 
        newinitd "${FILESDIR}/icingadb.initd" "${PN}"
+       systemd_dounit "${WORKDIR}/icingadb.service"
 
        keepdir /etc/icingadb
        keepdir /var/log/icingadb
```

**Comment by prometheanfire** [comment]
I don't think migrate should be a use flag (yet), it doesn't change external dependencies, I'd personally prefer to enable it and build it at all times, unless there's a good reason not to.  It's not a strong opinion, but it exists.

File: net-misc/icingadb/icingadb-1.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+RDEPEND="
+	acct-user/icinga
+	acct-group/icinga
+"
+
+DOCS=( config.example.yml doc README.md schema )
+
+src_prepare() {
+	default
+	sed -e "s|@EPREFIX@|${EPREFIX}|" "${FILESDIR}/icingadb.service" > "${WORKDIR}/icingadb.service" || die
```
Comment (juippis) [inline net-misc/icingadb/icingadb-1.2.0.ebuild:29]:
```suggestion
	sed -e "s|@EPREFIX@|${EPREFIX}|" "${FILESDIR}/icingadb.service" > "${T}/icingadb.service" || die
```

`${T}` is probably the more "legal" place to use here, but I don't think `${WORKDIR}` is **wrong** either. (if changed, needs to be updated in `src_install` too).

File: net-misc/icingadb/metadata.xml [hunk 2]
```diff
@@ -13,6 +13,9 @@
 		<email>proxy-maint@gentoo.org</email>
 		<name>Proxy Maintainers</name>
 	</maintainer>
+	<use>
+		<flag name="migrate">Install the icingadb-migrate tool.</flag>
```
Comment (juippis) [inline net-misc/icingadb/metadata.xml:17]: Nitpick but we (proxy-maint) generally don't use `.` at the end of a descriptive text like this:
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Descriptive_text

**Review by juippis — APPROVED** [review]
I also agree with @prometheanfire - I'd just always ship the mitigate binary since the modules are downloaded already. No **strong** opinion from me either, but the binary might come handy later on pre-installed.

The install size difference is quite notable though with and without the binary, although in both cases imho the overall size isn't too bad:

> Final size of installed tree:  27416 KiB (26.7 MiB)
> Final size of installed tree:  15188 KiB (14.8 MiB)

**Comment by sbraz** [comment]
Thanks for the review guys :) You're absolutely right about the `.` in the USE flag. I've pushed a version without the flag. I'll push it if it looks OK to you.

**Comment by hydrapolic** [comment]
Thanks for taking care of it @sbraz! :)

This makes https://github.com/gentoo/gentoo/pull/36438 obsolete then.

**Comment by sbraz** [comment]
> Thanks for taking care of it @sbraz! :)
> 
> This makes #36438 obsolete then.

Oh I didn't check open PRs, my bad. Everything is up-to-date now :) Feel free to ping me for future version bumps. I might not add myself to maintainers but I'm actively using Icinga.

**Pull request closed without merging**